src/ocean/text/convert/Formatter.d

```diff
@@ -381,9 +381,44 @@ private void handle (T) (T v, FormatInfo f, scope FormatterSink sf, scope ElemSi
     static if (is(T == typeof(null)))
         se("null", f);
 
-    // Cannot print enum member name in D1, so just print the value
-    else static if (is (T V == enum))
-        handle!(V)(v, f, sf, se);
+
+    // Print enum member name - If `switch statement` is available
+    else static if (is(T V == enum) && canSwitchOn!(V))
+    {
+        sw: switch (v)
+        {
+            foreach (member; __traits(allMembers, T))
+            {
+                case mixin("T.", member):
+                    sf(T.stringof ~ "." ~ member);
```

Comment:
`~` will allocate a new string. I suggest doing something like this instead:

```D
sf(T.stringof);
sf(".");
sf(member);
```

And please do this for all other calls to `sf` below where `~` was used.

Reply:
Note that, since they are all CT-known values, they get const-folded. But since it's not always obvious, that's why I asked for the `testNoAlloc` check. In any case your suggestion makes it explicit so 👍

Reply:
I revised it to your suggestion. Thank you for making such a good suggestion.